### PR TITLE
Use latest encryption subkey by default.

### LINKED ISSUE #1313: Choose the latest encryption-capable subkey by default.
## Description

Currently, when encrypting data to the primary key which is not able to encrypt, rnp chooses the first encryption-capable 'good' subkey. However, it could be more wise to choose the latest one.
See the discussion/report here:

https://thunderbird.topicbox.com/groups/e2ee/T12a8f43ab17206ec-M835b2ddbba1b0dfe3ec85cb5/tb-native-openpgp-implementation-does-not-use-newer-subkeys

### PR DESCRIPTION
**Pull request by ni4**
This PR fixes #1313 by defaulting to the latest encryption subkey, as GnuPG does.

Also as a side effect it:
- adds ffi-enc.cpp tests source file, and moves some encrypt/decrypt tests there for faster compilation.
- refactors some older tests for better readability

### CODE REVIEW

File: src/tests/ffi-enc.cpp [hunk 1]
```diff
@@ -0,0 +1,896 @@
...
+    assert_rnp_success(rnp_recipient_get_keyid(recipient, &keyid));
+    assert_non_null(keyid);
+    assert_string_equal(keyid, "8A05B89FAD5ADED1");
+    rnp_buffer_destroy(keyid);
+
+    rnp_op_verify_destroy(verify);
+    rnp_input_destroy(input);
+    rnp_output_destroy(output);
+
+    /* next pass */
```
Comment (rrrooommmaaa) [inline src/tests/ffi-enc.cpp:820]: Can we have a description how this pass differs, please?

File: src/tests/ffi-enc.cpp [hunk 2]
```diff
@@ -0,0 +1,896 @@
...
+    assert_rnp_success(rnp_recipient_get_keyid(recipient, &keyid));
+    assert_non_null(keyid);
+    assert_string_equal(keyid, "1ED63EE56FADC34D");
+    rnp_buffer_destroy(keyid);
+
+    rnp_op_verify_destroy(verify);
+    rnp_input_destroy(input);
+    rnp_output_destroy(output);
+
+    /* next pass */
```
Comment (rrrooommmaaa) [inline src/tests/ffi-enc.cpp:875]: Can we have a description how this pass differs and why it fails, please?

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]
Please provide some description of test data

**Review by antonsviridenko — APPROVED** [review]

File: src/tests/ffi-enc.cpp (hunk 1 above)
Comment (ni4) [reply src/tests/ffi-enc.cpp]: @rrrooommmaaa Sure, added description and re-pushed.

**Review by rrrooommmaaa — APPROVED** [review]
LGTM!

**Comment by ronaldtse** [comment]
@ni4 could you help rebase this and merge? Thanks!

**Comment by ni4** [comment]
@ronaldtse Sure, will do! This will need #1328 to be merged first to let macOS CI run smooth. Some sort of CI disaster this week )

**Comment by ronaldtse** [comment]
macOS is supposed to be fixed in #1328 , once that patch is brought over, macOS should pass.

**Comment by ni4** [comment]
Merging as we have 2 approvals and all-green CI.

**Pull request merged**